src/verilog/verilog_typecheck_expr.cpp

```diff
@@ -431,22 +434,21 @@ void verilog_typecheck_exprt::convert_system_function(
     if(argument.type().id()==ID_signedbv)
     {
       // remove
-      exprt tmp=argument;
-      expr.swap(tmp);
+      return argument;
     }
     else if(argument.type().id()==ID_unsignedbv)
     {
       exprt tmp(ID_typecast, argument.type());
       tmp.type().id(ID_signedbv);
       tmp.add_to_operands(std::move(argument));
       tmp.add_source_location()=expr.source_location();
-      expr.swap(tmp);
+      return tmp;
```

Comment:
Should this code be cleaned up as well? Like this:
````c++
typet new_type{argument.type()};
new_type.id(ID_signedbv);
typecast_exprt tmp{std::move(argument), std::move(new_type)};
tmp.add_source_location() = expr.source_location();
return std::move(tmp);
````

Reply:
done